pkg/config/utils/tags.go

```diff
@@ -24,5 +27,8 @@ func GetConfiguredTags(c config.Reader, includeDogstatsd bool) []string {
 	combined = append(combined, extraTags...)
 	combined = append(combined, dsdTags...)
 
+	// The aggregator sorts and removes duplicates in place. Pre-sorting part of the tags should
+	// improve the performances of the insertion sort in the aggregators.
```

Comment:
Not sure of this, since there are more tags added to these ones (tagger tags) and a sort is still necessary in the aggregator. Since these tags are unknown, nothing proves that it is creating a better scenario for the sort executed in the aggregator.

It would be great to assess it with micro-benchmarks or similar.

Reply:
I asked SMP to do some benchmarks. I just relied on the fact that we use an insertion sort internally which is more efficient when part of the slices are already pre-sorted. This place is definitely not the only place where we could add this.

Reply:
Discussed this in slack, our suggestion is to rebase from `main` to pick up more recent experiments and perhaps tailor an experiment to push larger sets of tags. Right now, for instance, most experiments stop at 50 tags. None of the tagsets [lading](https://github.com/datadog/lading) produces are sorted. Based on the most recent regression run Nov. 8 [here](https://app.datadoghq.com/dashboard/x9i-4yi-27x/smp-agent-regression-detector-capture-data?refresh_mode=paused&tpl_var_run-id%5B0%5D=8f692150-29b7-47a8-9761-b6f5f402d788&from_ts=1699440504515&to_ts=1699441048359&live=false) this change is neutral across the board. It will be instructive to compare profiles, see [this page](https://app.datadoghq.com/profiling/search?query=account%3Asingle-machine-performance%20runtime%3Ago%20service%3Adatadog-agent%20&my_code=disabled&viz=flame_graph&start=1702491839633&end=1703096639633&paused=false) for profiles collected from regression experiments in the past week. Comparison is possible by limiting to image tags, experiment names.

Reply:
After rebasing from main and re-running the experiments with the new _with_tags experiment added, conclusion remains the same: this change offers neutral results across the board. See [capture data](https://app.datadoghq.com/dashboard/x9i-4yi-27x/smp-agent-regression-detector-capture-data?refresh_mode=paused&tpl_var_exe%5B0%5D=agent&tpl_var_experiment%5B0%5D=uds_dogstatsd_to_api_with_dd_tags&tpl_var_run-id%5B0%5D=e12bcc48-e039-4047-856e-df8465f0e53f&view=spans&from_ts=1703641848353&to_ts=1703642352971&live=false) for more detail.